### PR TITLE
CMake: Include only the libraries required

### PR DESCRIPTION
**Pull request by hugueskamba**
Mbed OS now handles the dependencies.

Depends on https://github.com/ARMmbed/mbed-os/pull/14078 for OSPIF support.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -24,13 +24,37 @@ target_sources(${APP_TARGET}
         main.cpp
 )
 
+list(APPEND mbed_storage_libs)
+
+if("DATAFLASH" IN_LIST MBED_TARGET_LABELS)
```
Comment (0xc0170) [inline CMakeLists.txt:29]: This is actually good point for the migration path. As the old tools use "components_add", the new tools generate labels from "components_add" and we can include what is needed in examples. Please add  this note to our migration guide story so we do not forget

**Review by 0xc0170 — COMMENTED** [review]
One commit contains invalid character: `CMake: Include only the storage library needed`

**Comment by hugueskamba** [comment]
> One commit contains invalid character: `CMake: Include only the storage library needed`

Fixed.

File: CMakeLists.txt (hunk 1 above)
Comment (hugueskamba) [reply CMakeLists.txt:29]: `components_add` is to say that the code for that component should be included in the build.
What I added is to selectively include Mbed storage Libraries depending on what component is required by the application.

So `components_add` is still required.

**Review by rajkan01 — APPROVED** [review]
LGTM

**Pull request merged**